File: lib/go/IProtocol.go
//==============================================
/**
	@date:		2015:8:24  
	@file: 		IProtocol.go
	@author		zhangshuai
*/
//==============================================
package rpc

type  IProtocol interface
{
	WriteMsgBegin()
	WriteMsgEnd()
	/**  Writing functions. */
	
	WriteBool(b bool)
	
	WriteUInt8( i uint8)
	WriteInt8( i int8)
	
	WriteUInt16(i uint16) 
	WriteInt16(i int16 ) 
	
	WriteUInt32( i uint32) 
	WriteInt32( i int32)
	
	WriteInt64(i int64) 
	
	WriteFloat(f float32) 
	
	WriteString(str string) 
	
	/** Reading functions */
	ReadBool() (bool)
	
	ReadUInt8() (uint8)
	ReadInt8() (int8)
	
	ReadUInt16() (uint16)
	ReadInt16() (int16)
	
	ReadUInt32() (uint32)
	ReadInt32() (int32)
	
	ReadInt64() (int64)
	
	ReadFloat() (float32)
	
	ReadString() (string)
}

File: lib/csharp/IProtocol.cs
using System;
public interface IProtocol  
{
	void  writeMsgBegin();
	void  writeMsgEnd();
	/**  Writing voids. */
	
	void  writeBool(bool value);
	
	void  writeUInt8( byte i);
	void  writeInt8( sbyte i) ;

	void  writeUInt16(ushort i) ;
	void  writeInt16(short i ) ;
	
	void  writeUInt32( uint i);
	void  writeInt32( int i) ;
	
	void  writeInt64(long i);
	
	void  writeFloat(float f);
	
	void  writeString(string str);
	
	/** Reading voids */
	bool  readBool();
	
	byte  readUInt8();
	sbyte  readInt8();
	
	ushort  readUInt16();
	short  readInt16();
	
	uint  readUInt32();
	int  readInt32();
	
	long  readInt64();
	
	float  readFloat();
	
	string  readString();
}
